tests/queries/0_stateless/02131_skip_index_not_materialized.sql

```diff
@@ -0,0 +1,11 @@
+DROP TABLE IF EXISTS t_index_non_materialized;
+
+CREATE TABLE t_index_non_materialized (a UInt32) ENGINE = MergeTree ORDER BY tuple();
+
+INSERT INTO t_index_non_materialized VALUES (1);
+
+ALTER TABLE t_index_non_materialized ADD INDEX ind (a) TYPE set(1) GRANULARITY 1;
```

Comment:
Nice catch! Worth adding a check for minmax index too? (since it is a separate code branch)

Reply:
There already exists a check https://github.com/ClickHouse/ClickHouse/blob/4e6bf2456c8caa3fc718b537d64ae1b44d6e2ab7/src/Storages/MergeTree/MergeTreeIndexMinMax.cpp#L216-L223

 Added it to test.

Reply:
>There already exists a check

Yep, I meant adding it to test (since those branches can be modified independently)